### PR TITLE
Validate ActionBlock constructor inputs

### PR DESCRIPTION
**Pull request by yevhen**
## Summary
- fail fast when ActionBlock is configured with zero or negative parallelism
- reject null processing actions at construction time
- add regression tests for invalid constructor inputs

## Tests
- dotnet build
- DOTNET_ROLL_FORWARD=Major dotnet test -v minimal

### CODE REVIEW

**Review by yevhen — COMMENTED** [review]
One requested test coverage follow-up.

File: src/Simpipe.Net/Blocks/ActionBlock.cs [hunk 1]
```diff
@@ -139,7 +139,10 @@ public ActionBlock(
         BlockItemAction<T>? done = null,
         CancellationToken cancellationToken = default)
     {
-        this.action = action;
+        if (parallelism <= 0)
+            throw new ArgumentOutOfRangeException(nameof(parallelism), "Parallelism must be greater than 0.");
+
+        this.action = action ?? throw new ArgumentNullException(nameof(action));
```
Comment (yevhen) [inline src/Simpipe.Net/Blocks/ActionBlock.cs:145]: This adds a new `ArgumentNullException` path for `action`, but the PR only tests invalid `parallelism`. Could we add a constructor validation test for `action: null` as well?

**Review by yevhen — COMMENTED** [review]
Review note: please add missing test coverage for the new action null-validation path.
